chemprop/featurizers/atom.py

```diff
@@ -42,27 +42,27 @@ class MultiHotAtomFeaturizer(AtomFeaturizer):
     hybridizations: Sequence[HybridizationType] = field(default_factory=list)
     
     def __post_init__(self):
-        self.atomic_nums = {j: i for i, j in enumerate(self.atomic_nums)}
-        self.degrees = {i: i for i in self.degrees}
-        self.formal_charges = {j: i for i, j in enumerate(self.formal_charges)}
-        self.chiral_tags = {i: i for i in self.chiral_tags}
-        self.num_Hs = {i: i for i in self.num_Hs}
-        self.hybridizations = {ht: i for i, ht in enumerate(self.hybridizations)}
+        self._atomic_nums = {j: i for i, j in enumerate(self.atomic_nums)}
+        self._degrees = {i: i for i in self.degrees}
+        self._formal_charges = {j: i for i, j in enumerate(self.formal_charges)}
+        self._chiral_tags = {i: i for i in self.chiral_tags}
+        self._num_Hs = {i: i for i in self.num_Hs}
+        self._hybridizations = {ht: i for i, ht in enumerate(self.hybridizations)}
         self._subfeats: list[dict] = [
-            self.atomic_nums,
-            self.degrees,
-            self.formal_charges,
-            self.chiral_tags,
-            self.num_Hs,
-            self.hybridizations,
+            self._atomic_nums,
+            self._degrees,
+            self._formal_charges,
+            self._chiral_tags,
+            self._num_Hs,
+            self._hybridizations,
         ]
         subfeat_sizes = [
-            1 + len(self.atomic_nums),
-            1 + len(self.degrees),
-            1 + len(self.formal_charges),
-            1 + len(self.chiral_tags),
-            1 + len(self.num_Hs),
-            1 + len(self.hybridizations),
+            1 + len(self._atomic_nums),
+            1 + len(self._degrees),
+            1 + len(self._formal_charges),
+            1 + len(self._chiral_tags),
+            1 + len(self._num_Hs),
+            1 + len(self._hybridizations),
             1,
             1,
         ]
```

Comment:
I don't really understand this change- why do we need to store both the input arguments to the class and some private attributes if we never use the former _at all_? My point from earlier is that we should move this class from being defined as a `dataclass` to just being a regular class with an `__init__()` method. As it stands, we're no longer really using the functionality that comes with being a dataclass and just making the code more verbose